include/lsst/afw/math/LeastSquares.h

```diff
@@ -122,8 +122,8 @@ class LeastSquares {
         // the weird C++ syntax required for calling a templated member function
         // called "cast" in this context; see
         // http://eigen.tuxfamily.org/dox-devel/TopicTemplateKeyword.html
-        _getDesignMatrix() = design.asEigen().template cast<double>();
-        _getDataVector() = data.asEigen().template cast<double>();
+        _getDesignMatrix() = ndarray::asEigenMatrix(design).template cast<double>();
+        _getDataVector() = ndarray::asEigenMatrix(data).template cast<double>();
```

Comment:
I hope @TallJimbo will say more, but my understanding is that this makes a copy because the target is a (reference to) an `Eigen::MatrixXd`

Reply:
The assignment operation is where the (only) copy is made.  The call to `cast` is safe only because temporaries returned by `asEigenMatrix` are not destroyed until the end of their statements (see e.g. https://stackoverflow.com/questions/3041249/when-are-temporaries-created-as-part-of-a-function-call-destroyed).

Reply:
Ah, ok, I didn't realize the assignment copied the data. Maybe I've spent too long staring at `ndarray` and `afw::image`... :upside_down_face: